clap_blocks/src/querier.rs

```
@@ -0,0 +1,78 @@
+/// CLI config for querier configuration
+#[derive(Debug, Clone, PartialEq, clap::Parser)]
+pub struct QuerierConfig {
+    /// The number of threads to use for queries.
+    ///
+    /// If not specified, defaults to the number of cores on the system
+    #[clap(long = "--num-query-threads", env = "INFLUXDB_IOX_NUM_QUERY_THREADS")]
+    pub num_query_threads: Option<usize>,
+
+    /// gRPC address for the querier to talk with the ingester. For
+    /// example "http://127.0.0.1:8083" or
+    /// "http://10.10.10.1:8083,http://10.10.10.2:8083" for multiple
+    /// addresses.
+    ///
+    /// Note this is a workround for
+    /// <https://github.com/influxdata/influxdb_iox/issues/3996>
+    #[clap(
+        long = "--ingester-address",
+        env = "INFLUXDB_IOX_INGESTER_ADDRESSES",
+        default_value = "http://127.0.0.1:8083",
+        multiple_values = true,
+        use_value_delimiter = true
+    )]
+    pub ingester_addresses: Vec<String>,
```

One potential config error that can happen here (but that we have in other places of our stack as well) is that we would name the same ingester multiple times and the querier silently accepts it and potentially returns duplicate rows (because it won't perform any cross-ingester de-dup). If we wanna fix that then I would opt for bailing out instead of silently converting this vector to a set.

---

I will add some error checking

---

6e57fee1a2a84516f53c4ed9e46baf4e5f062291